pagerduty/provider.go

```diff
@@ -97,13 +101,26 @@ func Provider() *schema.Provider {
 	}
 
 	p.ConfigureFunc = func(d *schema.ResourceData) (interface{}, error) {
-		terraformVersion := p.TerraformVersion
```

Comment:
Note that the *only* thing that was happening in `ConfigureFunc` was normalizing the Terraform version. That should not be necessary when using `Provider.UserAgent` so I just moved the body of `providerConfigure` here. I didn't see a reason to have a separate function.